Sources/Subs-Db-mysql.php

```diff
@@ -825,7 +825,7 @@ function smf_db_insert($method = 'replace', $table, $columns, $data, $keys, $ret
 		$connection
 	);
 	
-	if(!empty($keys) && (count($keys) > 0) && $method == '' && $returnmode > 0)
+	if(!empty($keys) && (count($keys) > 0) && ($method === '' || $method === 'insert') && $returnmode > 0)
```

Comment:
In this case, it would be more efficient to test if `$queryTitle === 'INSERT'` rather than if `($method === '' || $method === 'insert')`